### DISCUSSION TITLE
Maintenance status of sqlboiler?

### DISCUSSION THREAD

**Discussion by agis** [aarondl/sqlboiler#903] [General]
Hello there @aarondl!

We're currently using sqlboiler in production and we've been happy so far - thanks for this great library! 💯 

That said, I've noticed some places were you state that sqlboiler is in maintenance mode ([1](https://github.com/volatiletech/sqlboiler/issues/134#issuecomment-663869226), [2](https://github.com/volatiletech/sqlboiler/issues/32#issuecomment-244830852)). Can you please elaborate on what this actually means for this project? (Particularly with regards to security issues, bugs and new features.)

Do you still recommend using SQLBoiler in production, or do you suggest some other alternative? Do you still use it in @volatiletech?

And one last off-topic question: `lib/pq` suggests moving on to `pgx` ([link](https://github.com/lib/pq#status)). What's your take on this? Are there any plans to move off from `lib/pq`?

Thanks in advance!

**Comment by aarondl** [comment]
The first link is correct (1), the second (2) is not. It's in maintenance mode but I do still fix bugs and users still submit both fixes and features which I accept. Security issues if there were any would be promptly dealt with.

I do recommend sqlboiler for production, I use it in production myself and it works fine. It does have problems around joins/eager loads and I'm not proud of the story there. That part desperately needs a rewrite but since I haven't felt the pain myself I have not done anything about it as it's a decent size project which will almost certainly be API breaking.

The only alternative that I'd be able to recommend is sqlc which is quite good until you hit its edge cases. The support for replacing types is really finicky and every once in a while you'll hit a query that you just can't use with it at all and have to write it raw which is pretty annoying. I think both sqlc and sqlboiler are good solutions, neither is perfect and it comes down to what users like.

The whole lib/pq vs pgx simply doesn't matter, if someone wanted to do it sure but the fact of the matter is it's only used to do a couple basic queries to see what the shape of your database is. When you actually use sqlboiler in your application you define the database type via the `boil.(Context)Executor` interface so you're free to use pgx if you prefer it.

**Reply by agis** [reply]
Awesome, thanks for clarifying!
